bot/moderation/moderation.py

```diff
@@ -731,15 +732,17 @@ async def purge_messages(self, ctx: commands.Context, channel: Optional[discord.
 
         confirmation_embed = _build_purge_confirmation_embed(purge_channel, amount)
         is_confirmed = await self._send_confirmation_dialog(ctx, confirmation_embed)
-
+        is_posted_in_purge_channel = purge_channel is ctx.channel
         if is_confirmed:
-            deleted_messages = await purge_channel.purge(limit=amount + 1)
-            await purge_channel.send('**Ich habe __{0} Nachrichten__ erfolgreich gelöscht.**'
-                                     .format(len(deleted_messages) - 1), delete_after=constants.TIMEOUT_INFORMATION)
+            deleted_messages = await purge_channel.purge(limit=amount + 1 if is_posted_in_purge_channel else amount)
+            await purge_channel.send(
+                '**Ich habe __{0} Nachrichten__ erfolgreich gelöscht.**'.format(len(deleted_messages)),
+                delete_after=constants.TIMEOUT_INFORMATION)
             log.info("SAM deleted %s messages in [#%s]", len(deleted_messages), purge_channel)
 
+            details = f"Deleted {len(deleted_messages)} mesages in channel #{purge_channel}"
```

Comment:
How about we add an "amount" and "channel" field instead of the one named "details"?
I would also like it more if the channel was a real mention (`channel.mention()`) instead of simple text.

Reply:
I intentionally left it at details, because we need it for different things and it should be a general mod embed in this method. 
Otherwise we'd have 100s of arguments and 100s of if else for the different fields. 
I'd say we could replace all of them with a dict if you want to.

Channel mention my bad, I forgot to change that.